src/fosslight_binary/_binary_dao.py

```
@@ -2,185 +2,281 @@
+    results_by_id = {}
     try:
-        conn = psycopg2.connect(connection_string)
-        cur = conn.cursor()
+        for chunk_start in range(0, len(items_payload), _CHUNK_SIZE):
+            chunk = items_payload[chunk_start: chunk_start + _CHUNK_SIZE]
+            response = _post_binary_match(base_url, token, chunk)
+            if response is None:
+                return bin_info_list, _cnt_auto_identified
+            for result in response.get("results", []):
+                results_by_id[str(result.get("id"))] = result
     except Exception as ex:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)